cloud/services/securitygroups/client.go

```diff
@@ -71,6 +86,33 @@ func (ac *AzureClient) CreateOrUpdate(ctx context.Context, resourceGroupName str
 	return err
 }
 
+// createOrUpdatePreparer prepares the CreateOrUpdate request.
+func (ac *AzureClient) createOrUpdatePreparer(ctx context.Context, resourceGroupName string, networkSecurityGroupName string, parameters network.SecurityGroup, etag string) (*http.Request, error) {
```

Comment:
This content of this func should probably be the following, not building the Preparer from scratch:
```go
// ac.CreateOrUpdatePreparer would expose ac.securitygroups.CreateOrUpdatePreparer 
req, err := ac.CreateOrUpdatePreparer(ctx, resourceGroupName, networkSecurityGroupName, parameters)
if err != nil {
    return nil, err
}

req.Header.Add("If-Match", etag)
```

Reply:
I opened https://github.com/Azure/azure-sdk-for-go/issues/8917 to ask about plans to build this into the Go SDK.

Reply:
@devigned Removed the preparer as suggested.